### PR TITLE
Add metadata headers when uploading objects

### PR DESCRIPTION
**Pull request by neowulf33**
This PR needs another PR merged - https://github.com/akka/alpakka/pull/140

### CODE REVIEW

File: s3/src/test/scala/akka/stream/alpakka/s3/impl/S3NoMock.scala [hunk 1]
```diff
@@ -16,26 +14,25 @@ import org.scalatest.{BeforeAndAfterAll, FlatSpecLike, Matchers}
...
   it should "upload with real credentials" ignore {
 
     val source: Source[ByteString, Any] = Source(ByteString(objectValue) :: Nil)
+    //val source: Source[ByteString, Any] = FileIO.fromPath(Paths.get("/tmp/IMG_0470.JPG"))
 
-    val result = source.runWith(S3Client().multipartUpload(bucket, objectKey, contentType = ContentTypes.`text/plain(UTF-8)`))
+    val result = source.runWith(S3Client().multipartUpload(bucket, objectKey, metaHeaders = MetaHeaders(metaHeaders)))
 
-    val multipartUploadResult = Await.ready(result, 5.seconds).futureValue
+    val multipartUploadResult = Await.ready(result, 90.seconds).futureValue
```
Comment (patriknw) [inline s3/src/test/scala/akka/stream/alpakka/s3/impl/S3NoMock.scala]: just curious, what is the materialized value (result)? A Future of CompleteMultipartUploadResult? Looks like a Future[Future] here?

Is 90 seconds realistic (needed)?

Comment (neowulf33) [inline s3/src/test/scala/akka/stream/alpakka/s3/impl/S3NoMock.scala]: I was testing large file uploads here, so I had to increase the timeout.

**Comment by neowulf33** [comment]
I have updated the commit message.

File: s3/src/main/scala/akka/stream/alpakka/s3/impl/HttpRequests.scala [hunk 2]
```diff
@@ -43,11 +53,15 @@ private[alpakka] object HttpRequests {
...
       conf: S3Settings): Future[HttpRequest] = {
 
+    //Do not let the start PartNumber,ETag and the end PartNumber,ETag be on different lines
+    //  They tend to get split when this file is formatted by IntelliJ
     val payload = <CompleteMultipartUpload>
-                    {
-                      parts.map { case (partNumber, etag) => <Part><PartNumber>{ partNumber }</PartNumber><ETag>{ etag }</ETag></Part> }
-                    }
-                  </CompleteMultipartUpload>
+      {parts.map { case (partNumber, etag) => <Part>
```
Comment (filosganga) [inline s3/src/main/scala/akka/stream/alpakka/s3/impl/HttpRequests.scala]: Are you sure this formatting will not break the AWS signature? I have experienced that in the `s3stream` original project.

Comment (neowulf33) [inline s3/src/main/scala/akka/stream/alpakka/s3/impl/HttpRequests.scala]: :) Yes, in fact this has caused me a lot of grief! Still not the best solution as IntelliJ will just wreck this file.  I have reverted this change as well as implemented a more permanent solution but is specific to IntelliJ in the next commit.

**Comment by patriknw** [comment]
@neowulf33 Can you rebase this so that it can be included in today's release. If you are in different time zone or not able to do it now I can do it for you. Let me know, if you see this.

**Comment by patriknw** [comment]
Refs https://github.com/akka/alpakka/issues/132

**Comment by patriknw** [comment]
Superseded by https://github.com/akka/alpakka/pull/146

**Comment by neowulf33** [comment]
Sorry @patriknw - I didn't see the message in time. Thanks for merging these PRs

**Pull request closed without merging**